### ISSUE TITLE
initialState is not being invoked

### ISSUE THREAD

**Issue by alan-agius4** [ngrx/store#273]
initialState is not being invoked when it's a function

```ts
StoreModule.provideStore(_provideRootReducer, _initialState),


export function _provideRootReducer(state: any, action: any) {
	return startupContext.appContext.isDebug
		? compose(storeLogger(), combineReducers)(appReducer)(state, action)
		: combineReducers(appReducer)(state, action);
}

export function _initialState() {
	return {
		appContext: startupContext.appContext,
		market: {
			currentMarket: startupContext.market,
			markets: startupContext.marketList
		}
	};
}

**Comment by metanav** [comment]
Does invoking it  in  provideStore not work? 

```
StoreModule.provideStore(_provideRootReducer, _initialState())
```

**Comment by alan-agius4** [comment]
Not wtih angular2 AOT, you cannot invoke functions

**Comment by metanav** [comment]
After #269 do we need to initialize ourselves?

**Comment by alan-agius4** [comment]
@metanav, i think you referenced a wrong PR. 

What the function invocation does is to return the object aka values for the initial state. There shouldn't be any changes from you jor any users which are not using AOT. If they arr using AOT, they now have an clean option how to provide a dynamic initial state

**Comment by bortexz** [comment]
Is there any workaround towards this? I use angular-cli and I have reducers as a function like: 

```javascript
export const authReducer: ActionReducer<AuthenticationState> =
(state: AuthenticationState = initialState, action: Action) => {
  console.log(action);
  switch (action.type) {
    case AuthenticationActions.LOGIN: {
      return null;
    }

    case AuthenticationActions.LOGIN_SUCCESS: {
      return action.payload;
    }

    case AuthenticationActions.LOGIN_ERROR: {
      return null;
    }

    case AuthenticationActions.LOGOUT: {
      return null;
    }

    default: {
      return state;
    }
  }
};

...

export function reducers () {
  return compose(localStorageSync(['authenticated'], true), combineReducers)({
    authenticated: authReducer
  });
};

...

imports: [
  StoreModule.provideStore(reducers)
]
```

I have effects setted up, and they work, but the reducer is never called, neither with initialState or after when I dispatch functions. Any idea? It may be related to this?

**Comment by florent1933** [comment]
@bertofer : I have exactly the same problem if you find a workaround it would be great 👍

**Comment by alan-agius4** [comment]
I did a PR for this but no response so far

https://github.com/ngrx/store/pull/272

**Comment by florent1933** [comment]
@alan-agius4 : I am going to to test your PR. Thanks a lot :)

**Comment by bortexz** [comment]
Hi @florent1933 , I could make it work by calling the reducer with state and action in the exported reducers function, like this:

```javascript
export function reducers (state, action) {
  // return composed(state, action);
  return compose(localStorageSync(['authenticated'], true), combineReducers)({
    authenticated: authReducer
  })(state, action);
};
```

Being this the function provided to `StoreModule.provideStore(reducers)`. Although I am not sure it belongs to this issue, for me the reducer was never being called, so I will probably open a new issue on this when I investigate more

**Comment by florent1933** [comment]
@alan-agius4 : I think our issue is not related to yours so your PR didn't fix it. 

@bertofer : Great, it's working 👍 Congrat's!

**Comment by bfricka** [comment]
@bertofer This works if you're aware ahead of time of the keys you want to populate. However, this makes it static. In order be dynamic, what you really want is to be able to do something like:

```ts
export const initialState = {router: initialRouterState, ...someExternalInitialStateObject};
```
Amusingly, AoT will compile like this, but what it compiles to is a train wreck. Here is the compiled result of this expression in AoT:
```ts
this.__INITIAL_STATE_65 = { router: { path: 0 } };
```
You see, because AoT is requiring 100% static symbols, the only way to approach this is with functions, which this project does not support. Without this, I have no choice but to fork the project and implement it myself until upstream catches up.
